app/src/main/java/com/concordia/mcga/lib/BuildingBackdropBottomSheetBehavior.java

```diff
@@ -0,0 +1,123 @@
+package com.concordia.mcga.lib;
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.design.widget.CoordinatorLayout;
+import android.support.v4.widget.NestedScrollView;
+import android.util.AttributeSet;
+import android.view.View;
+
+import java.lang.ref.WeakReference;
+
+/**
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~      http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~
+ ~ https://github.com/miguelhincapie/CustomBottomSheetBehavior
+ */
+
+/**
+ * This class will link the Backdrop element (that can be anything extending View) with a
+ * NestedScrollView (the dependency). Whenever dependecy is moved, the backdrop will be moved too
+ * behaving like parallax effect.
+ *
+ * The backdrop need to be <bold>into</bold> a CoordinatorLayout and <bold>before</bold>
+ * {@link BuildingBottomSheetInfo} in the XML file to get same behavior like Google Maps.
+ * It doesn't matter where the backdrop element start in XML, it will be moved following
+ * Google Maps's parallax behavior.
+ * @param <V>
+ */
+public class BuildingBackdropBottomSheetBehavior<V extends View> extends CoordinatorLayout.Behavior<V> {
+    /**
+     * To avoid using multiple "peekheight=" in XML and looking flexibility allowing {@link BuildingBottomSheetInfo#mPeekHeight}
+     * get changed dynamically we get the {@link NestedScrollView} that has
+     * "app:layout_behavior=" {@link BuildingBottomSheetInfo} inside the {@link CoordinatorLayout}
+     */
+    private WeakReference<BuildingBottomSheetInfo> mBottomSheetBehaviorRef;
+    /**
+     * Following {@link #onDependentViewChanged}'s docs mCurrentChildY just save the child Y
+     * position.
+     */
+    private int mCurrentChildY;
+
+    public BuildingBackdropBottomSheetBehavior(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    @Override
+    public boolean layoutDependsOn(CoordinatorLayout parent, View child, View dependency) {
```

Comment:
I imported this class, ideally it needs to be a .jar but I couldn't get it to work that way